encord/objects/classification_instance.py

```diff
@@ -166,6 +166,17 @@ def set_for_frames(
             self._parent._add_frames_to_classification(self.ontology_item, frames_list)
             self._parent._add_to_frame_to_hashes_map(self, frames_list)
 
+    def set_frame_data(self, frame_data: FrameData, frames: Frames) -> None:
+        frames_list = frames_class_to_frames_list(frames)
+
+        for frame in frames_list:
+            self._frames_to_data[frame] = frame_data
```

Comment:
did you check if update operations function correctly with the same frame_data instances assigned to multiple frames? (the update method you touched below looks like it is called by the public set_for_frames function).

Reply:
On the public set, there's always one frame in the frame list T_T

Reply:
it may be worth removing the in-place update function and making this object properly immutable to prevent any accidental multi-updates. (maybe i'm overthinking it).